spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/properties/RateLimitType.java

```diff
@@ -63,14 +64,30 @@ public boolean apply(HttpServletRequest request, Route route, RateLimitUtils rat
         }
 
         @Override
-        public String key(HttpServletRequest request, Route route, RateLimitUtils rateLimitUtils) {
+        public String key(HttpServletRequest request, Route route, RateLimitUtils rateLimitUtils, String matcher) {
             return Optional.ofNullable(route).map(Route::getPath).orElse(StringUtils.EMPTY);
         }
     },
+
+    /**
+     * Rate limit policy considering the authenticated user's role.
+     */
+    ROLE {
+        @Override
+        public boolean apply(HttpServletRequest request, Route route, RateLimitUtils rateLimitUtils, String matcher) {
+            return rateLimitUtils.getUserRoles().contains(matcher.toUpperCase());
+        }
+
+        @Override
+        public String key(HttpServletRequest request, Route route, RateLimitUtils rateLimitUtils, String matcher) {
+            return matcher;
```

Comment:
```suggestion
            return matcher.toUpperCase();
```
Since this is the way we check it no?

Reply:
I tried it and didn't work, I don't think we use the key to test it